datalad/support/path.py

```diff
@@ -48,3 +51,18 @@ def wrapped(*args, **kwargs):
 relpath = op.relpath
 realpath = _get_unicode_robust_version(op.realpath)
 sep = op.sep
+
+
+def robust_abspath(p):
+    """A helper which would not fail if p is relative and we are in non-existing directory
+
+    It will rely on getpwd, which would rely on $PWD env variable to report
+    the path.  Desired for improved resilience during e.g. reporting as in
+    https://github.com/datalad/datalad/issues/2787
+    """
+    try:
+        return abspath(p)
+    except OSError as exc:
+        if "No such" in str(exc) and not isabs(p):
```

Comment:
Eh I dislike relying on matching a particular string in the exception message, though I know there are already lots of case where we do it in the codebase.  In this case, couldn't you `getpwd()` and raise unless it *doesn't* exist and `not isabs(p)`?

Reply:
Ok, will do shortly

Reply:
Although it would make reraising original exception more cumbersome iirc